ztp/siteconfig-generator/siteConfig/siteConfigBuilder_test.go

```diff
@@ -0,0 +1,160 @@
+        rootDeviceHints:
+          hctl: "1:2:0:0"
+        cpuset: "2-19,22-39"
+        nodeNetwork:
+          interfaces:
+            - name: eno1
+              macAddress: "00:00:00:01:20:30"
+          config:
+            interfaces:
+              - name: eno1
+                macAddress: "00:00:00:01:20:30"
+                type: ethernet
+                ipv4:
+                  enabled: true
+                  dhcp: false
+`
+
+func Test_grtManifestFromTemplate(t *testing.T) {
+	tests := []struct {
+		template      string
+		data          interface{}
+		expectFn      string
+		expectContent string
+	}{{
+		template:      "testdata/good.yaml.tmpl",
+		data:          struct{ Value string }{"values"},
+		expectFn:      "role-good.yaml",
+		expectContent: "rendered-role: values\n",
+	}, {
+		template: "testdata/parse_failure.yaml.tmpl",
+	}, {
+		template: "testdata/execution_failure.yaml.tmpl",
+	}, {
+		template: "testdata/empty.yaml.tmpl",
+	}}
+	// Cannot test bad filename because that causes a panic
+	scb := SiteConfigBuilder{}
+	for _, test := range tests {
+		fn, content, _ := scb.getManifestFromTemplate(test.template, "role", test.data)
+		assert.Equal(t, test.expectFn, fn)
+		assert.Equal(t, test.expectContent, content)
+	}
+}
+
+func Test_siteConfigBuildValidation(t *testing.T) {
+
+	sc := SiteConfig{}
+	err := yaml.Unmarshal([]byte(siteConfigTest), &sc)
+	assert.Equal(t, err, nil)
+
+	scBuilder, _ := NewSiteConfigBuilder()
+	// Set empty cluster Name
+	sc.Spec.Clusters[0].ClusterName = ""
+	_, err = scBuilder.Build(sc)
+	assert.Equal(t, err, errors.New("Error: Missing cluster name at site test-site"))
+
+	// Set invalid network type
+	sc.Spec.Clusters[0].ClusterName = "cluster1"
+	sc.Spec.Clusters[0].NetworkType = "invalidNetworkType"
+	_, err = scBuilder.Build(sc)
+	assert.Equal(t, err, errors.New("Error: networkType must be either OpenShiftSDN or OVNKubernetes test-site/cluster1"))
+
+	// Set repeated cluster names
+	sc.Spec.Clusters[0].NetworkType = "OVNKubernetes"
+	sc.Spec.Clusters = append(sc.Spec.Clusters, sc.Spec.Clusters[0])
+	scBuilder.setLocalExtraManifestPath("../../source-crs/extra-manifest")
+	_, err = scBuilder.Build(sc)
+	assert.Equal(t, err, errors.New("Error: Repeated Cluster Name test-site/cluster1"))
+}
+
+func Test_siteConfigBuildExtraManifest(t *testing.T) {
+	sc := SiteConfig{}
+	err := yaml.Unmarshal([]byte(siteConfigTest), &sc)
+	assert.Equal(t, err, nil)
+
+	scBuilder, _ := NewSiteConfigBuilder()
+
+	// Expect to fail as the localExtraManifest path is in its default value
+	_, err = scBuilder.Build(sc)
+	assert.NotEqual(t, err, nil)
+	assert.Equal(t, strings.Contains(err.Error(), "no such file or directory"), true)
+
+	// Set the local extra manifest path to cnf-feature-deploy/ztp/source-crs/extra-manifest
+	scBuilder.setLocalExtraManifestPath("../../source-crs/extra-manifest")
+	// Setting the networkType to its default value
+	sc.Spec.Clusters[0].NetworkType = "OVNKubernetes"
```

Comment:
Why is this required?

Reply:
cause we override the NetworkType to include the annotation [L63](https://github.com/serngawy/cnf-features-deploy/blob/newSiteconfig/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L63)

Reply:
I didn't realize that!  That seems like a very bad thing to do, but I don't think we should change it today :)